crates/primitives/src/peer.rs

```diff
@@ -1,10 +1,32 @@
 use enr::Enr;
 use reth_rpc_types::NodeRecord;
-use secp256k1::SecretKey;
+use secp256k1::{PublicKey, SecretKey};
 use std::{net::IpAddr, str::FromStr};
+
 // Re-export PeerId for ease of use.
 pub use reth_rpc_types::PeerId;
 
+/// Converts a [secp256k1::PublicKey] to a [PeerId] by stripping the
+/// SECP256K1_TAG_PUBKEY_UNCOMPRESSED tag and storing the rest of the slice in the [PeerId].
+#[inline]
+pub fn pk2id(pk: &PublicKey) -> PeerId {
+    PeerId::from_slice(&pk.serialize_uncompressed()[1..])
+}
+
+/// Converts a [PeerId] to a [secp256k1::PublicKey] by prepending the [PeerId] bytes with the
+/// SECP256K1_TAG_PUBKEY_UNCOMPRESSED tag.
+#[inline]
+pub fn id2pk(id: PeerId) -> Result<PublicKey, secp256k1::Error> {
+    // NOTE: B512 is used as a PeerId because 512 bits is enough to represent an uncompressed
+    // public key.
+    let mut s = [0u8; 65];
+    // SECP256K1_TAG_PUBKEY_UNCOMPRESSED = 0x04
+    // see: https://github.com/bitcoin-core/secp256k1/blob/master/include/secp256k1.h#L211
+    s[0] = 4;
```

Comment:
these don't need to be magic numbers, see <https://github.com/paradigmxyz/reth/pull/7336>

Reply:
added a const. did not export since ideally this is the only place where it's needed